resolver.go

```diff
@@ -373,11 +391,13 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			}()
 			err = co.WriteMsg(query)
 			if err != nil {
+				r.concurrentQueryDec()
```

Comment:
`r.concurrentQueryDec()` is executed either if err == nil (line 400) or if err != nil (here).
Therefore, shouldn't we move it before the call `err = co.WriteMsg(query)` ?